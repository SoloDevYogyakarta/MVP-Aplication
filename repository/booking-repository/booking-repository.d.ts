import { BookingInstance } from '../../database/entities/services/booking-entity/booking-entity';
export declare class BookingRepository {
    private readonly logger;
    findAll(query: object, type: string): Promise<BookingInstance[]>;
    findOne(id: number): Promise<BookingInstance>;
}
