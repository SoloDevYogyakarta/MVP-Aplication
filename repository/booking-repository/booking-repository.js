"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var BookingRepository_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingRepository = void 0;
const common_1 = require("@nestjs/common");
const booking_entity_1 = require("../../database/entities/services/booking-entity/booking-entity");
let BookingRepository = BookingRepository_1 = class BookingRepository {
    constructor() {
        this.logger = new common_1.Logger(BookingRepository_1.name);
    }
    async findAll() {
        this.logger.log(BookingRepository_1.name);
        return await booking_entity_1.bookingEntity.findAll();
    }
    async findOne(id) {
        this.logger.log(BookingRepository_1.name);
        return await booking_entity_1.bookingEntity.findOne({ where: { id } });
    }
};
exports.BookingRepository = BookingRepository;
exports.BookingRepository = BookingRepository = BookingRepository_1 = __decorate([
    (0, common_1.Injectable)()
], BookingRepository);
//# sourceMappingURL=booking-repository.js.map