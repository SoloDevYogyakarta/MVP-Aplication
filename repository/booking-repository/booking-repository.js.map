{"version":3,"file":"booking-repository.js","sourceRoot":"","sources":["../../../src/repository/booking-repository/booking-repository.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAAoD;AACpD,sDAAmD;AACnD,mGAGwE;AAGjE,IAAM,iBAAiB,yBAAvB,MAAM,iBAAiB;IAAvB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,mBAAiB,CAAC,IAAI,CAAC,CAAC;IAgB/D,CAAC;IAdC,KAAK,CAAC,OAAO,CAAC,KAAa,EAAE,IAAY;QACvC,IAAI,KAAK,GAAG,EAAE,CAAC;QACf,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,mBAAiB,CAAC,IAAI,CAAC,CAAC;QACxC,KAAK,GAAG,IAAA,eAAM,EAAC,KAAK,EAAE,IAAI,CAAC,CAAC;QAC5B,OAAO,MAAM,8BAAa,CAAC,OAAO,CAAC;YACjC,KAAK,EAAE,KAAK;YACZ,KAAK,EAAE,CAAC,CAAC,IAAI,EAAE,KAAK,CAAC,CAAC;SACvB,CAAC,CAAC;IACL,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,mBAAiB,CAAC,IAAI,CAAC,CAAC;QACxC,OAAO,MAAM,8BAAa,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;IACxD,CAAC;CACF,CAAA;AAjBY,8CAAiB;4BAAjB,iBAAiB;IAD7B,IAAA,mBAAU,GAAE;GACA,iBAAiB,CAiB7B"}