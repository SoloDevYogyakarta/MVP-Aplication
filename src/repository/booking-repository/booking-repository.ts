import { Injectable, Logger } from '@nestjs/common';
import {
  bookingEntity,
  BookingInstance,
} from '../../database/entities/services/booking-entity/booking-entity';

@Injectable()
export class BookingRepository {
  private readonly logger = new Logger(BookingRepository.name);

  async findAll(): Promise<BookingInstance[]> {
    this.logger.log(BookingRepository.name);
    return await bookingEntity.findAll();
  }

  async findOne(id: number): Promise<BookingInstance> {
    this.logger.log(BookingRepository.name);
    return await bookingEntity.findOne({ where: { id } });
  }
}
