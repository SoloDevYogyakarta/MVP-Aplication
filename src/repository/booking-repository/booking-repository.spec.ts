import { BookingRepository } from './booking-repository';

describe('BookingRepository', () => {
  let repository: BookingRepository;

  beforeEach(() => {
    repository = new BookingRepository();
  });

  it('should to be defined', () => expect(repository).toBeDefined());

  it('render correctly', () => expect(repository).toMatchSnapshot());
});
