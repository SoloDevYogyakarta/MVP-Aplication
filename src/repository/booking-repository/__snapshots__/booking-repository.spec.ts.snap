// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingRepository render correctly 1`] = `
BookingRepository {
  "logger": Logger {
    "context": "BookingRepository",
    "options": {},
  },
}
`;
